packages/pulse-core/tests/state.test.ts

```diff
@@ -1,8 +1,245 @@
 import Pulse from '../lib';
 
 const App = new Pulse();
-const MyState = App.State(true);
+const BooleanState = App.State<boolean>(true);
+const StringState = App.State<string>('Hello Pulse!');
+const ObjectState = App.State<{ days: Days }>({ days: { monday: true, wednesday: true } });
+const NullState = App.State(null);
 
-test('State can be created', () => {
-  expect(MyState.value).toBe(true);
+interface Days {
+  monday?: boolean;
+  tuesday?: boolean;
+  wednesday?: boolean;
+  thursday?: boolean;
+  friday?: boolean;
+  saturday?: boolean;
+  sunday?: boolean;
+}
+
+//.value | Provides the current value (read-only)
+
+test('State can be created and retrieved', () => {
```

Comment:
We should wrap tests in a describe() for each set of tests. (e.g. here, use `describe('State', () => { ... })`) and have each of these `test()` methods contain the method being tested (in this case, `test('.value | Provides current value')` or something along those lines.